### DISCUSSION TITLE
capture the output of a bash command

### DISCUSSION THREAD

**Discussion by nickkaranatsios** [apple/pkl#620] [Q&A] [answered]
Is it possible to capture the output of a bash command and store it in a pkl variable. For example str = "`date '%s'`"
I guess is it possible to run a bash command from pkl.
Thank you in advance.

**Comment by HT154** [comment, marked as answer]
It's not possible to do this as you've described, but you may be able to use a [`prop:` resource](https://pkl-lang.org/main/current/language-reference/index.html#resources) in this specific case.

In your Pkl module:
```pkl
str = read("prop:timestamp")
```

Then when evaluating:
```
pkl eval path/to/module.pkl -p timestamp=$(date %s)
```

**Reply by nickkaranatsios** [reply]
Thanks for your reply I would try your suggestion.
